log/go.mod

```diff
@@ -3,13 +3,17 @@ module go.opentelemetry.io/otel/log
 go 1.20
 
 require (
+	github.com/go-logr/logr v1.4.1
```

Comment:
Is this dependency necessary to have in the logging API? Can we test the API a different way?

Reply:
Given we are logging within the API, if this isn't directly required it will always be indirectly required given the OTel logging in `internal/global` depends on it.

We could try to provide a pretty large refactor to the `internal/global` logging library that would have it accept a `LogSink` instead of the concrete `Logger` type. That would mean this could pass an interface without having to call `logr.New` to create a new `Logger` struct. However, this would only move the requirement to indirect instead of direct still.

Reply:
got it, thanks